common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		int timesY = (int) Math.floor(y / v);
+		int remainderX = x % u;
+		int remainderY = y % v;
+		for (int i = 0; i <= timesY; i++) {
+			for (int j = 0; j <= timesX; j++) {
+				if (j == timesX && i == timesY) {
+					drawStretchingSquare(x0 + (j * u), y0 + (i * v), x0 + (j * u) + remainderX, y0 + (i * v) + remainderY, z, u0, v0, u0 + remainderX, v0 + remainderY, blend, color);
+				} else if (j == timesX) {
+					drawStretchingSquare(x0 + (j * u), y0 + (i * v), x0 + (j * u) + remainderX, y0 + ((i + 1) * v), z, u0, v0, u0 + remainderX, v1, blend, color);
+				} else if (i == timesY) {
+					drawStretchingSquare(x0 + (j * u), y0 + (i * v), x0 + ((j + 1) * u), y0 + (i * v) + remainderY, z, u0, v0, u1, v0 + remainderY, blend, color);
+				} else {
+					drawStretchingSquare(x0 + (j * u), y0 + (i * v), x0 + ((j + 1) * u), y0 + ((i + 1) * v), z, u0, v0, u1, v1, blend, color);
+				}
+			}
+		}
+	}
+
+	public static void drawBoxedCenteredString(Minecraft mc, String text, int x, int y, int z) {
+		int width = mc.fontRenderer.getStringWidth(text);
+		int x1 = x - (width / 2 + 1);
+		int x0 = x1 - 4;
+		int x2 = x + (width / 2 + 1);
+		int x3 = x2 + 4;
+		int y0 = y;
+		int y1 = y0 + 4;
+		int y2 = y1 + 10;
+		int y3 = y2 + 4;
+		int u0 = 112;
+		int v0 = 32;
+		int u1 = 116;
+		int v1 = 36;
+		int u2 = 124;
+		int v2 = 44;
+		int u3 = 128;
+		int v3 = 48;
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(0.0F, 0.0F, z);
+		GlStateManager.translate(0.0F, 0.0F, 100.0F);
+		drawCenteredStringStatic(mc.fontRenderer, text, x, y + 5, 0xFFFFFF);
+		GlStateManager.translate(0.0F, 0.0F, -100.0F);
+		GlStateManager.enableAlpha();
+		mc.renderEngine.bindTexture(GUI_BOOK_TEXTURE);
+		// Background
+		drawRepeatingSquare(x1, y1, x2, y2, z, u1, v1, u2, v2, false);
+		// Corners: TopLeft, TopRight, BottomLeft & BottomRight
+		drawStretchingSquare(x0, y0, x1, y1, z, u0, v0, u1, v1, true);
+		drawStretchingSquare(x2, y0, x3, y1, z, u2, v0, u3, v1, true);
+		drawStretchingSquare(x0, y2, x1, y3, z, u0, v2, u1, v3, true);
+		drawStretchingSquare(x2, y2, x3, y3, z, u2, v2, u3, v3, true);
+		// Edges: Top, Bottom, Left & Right
+		drawStretchingSquare(x1, y0, x2, y1, z, u1, v0, u2, v1, true);
+		drawStretchingSquare(x1, y2, x2, y3, z, u1, v2, u2, v3, true);
+		drawStretchingSquare(x0, y1, x1, y2, z, u0, v1, u1, v2, true);
+		drawStretchingSquare(x2, y1, x3, y2, z, u2, v1, u3, v2, true);
+		GlStateManager.disableAlpha();
+		GlStateManager.popMatrix();
+	}
+
+	public static void drawCenteredStringStatic(FontRenderer fontRendererIn, String text, int x, int y, int color) {
+		fontRendererIn.drawStringWithShadow(text, (float) (x - fontRendererIn.getStringWidth(text) / 2), (float) y, color);
+	}
+
+	boolean equals(SavedTab a, SavedTab b) {
+		return a.page.getPage() == b.page.getPage() && a.page.getChapter() == b.page.getChapter() && a.page.getDivision() == b.page.getDivision() && a.state == b.state;
+	}
+
+	public class MenuItemsDivision {
```

Comment:
The file has 600 lines until this point. Please consider using multiple files for additional classes. Also consider the visibility of those classes. Do they all have to be *public*?

Reply:
Added to the Glo Board